worker/src/lib.rs

```diff
@@ -0,0 +1,592 @@
+    internal: &'a lv2_sys::LV2_Worker_Schedule,
+    phantom: PhantomData<*const P>,
+}
+
+unsafe impl<'a, P> UriBound for Schedule<'a, P> {
+    const URI: &'static [u8] = lv2_sys::LV2_WORKER__schedule;
+}
+
+unsafe impl<'a, P> Feature for Schedule<'a, P> {
+    unsafe fn from_feature_ptr(feature: *const c_void, class: ThreadingClass) -> Option<Self> {
+        if class == ThreadingClass::Audio {
+            (feature as *const lv2_sys::LV2_Worker_Schedule)
+                .as_ref()
+                .map(|internal| Self {
+                    internal,
+                    phantom: PhantomData::<*const P>,
+                })
+        } else {
+            panic!("The Worker Schedule feature is only allowed in the audio threading class");
+        }
+    }
+}
+
+impl<'a, P: Worker> Schedule<'a, P> {
+    /// Request the host to call the worker thread.
+    ///
+    /// This method should be called from `run()` context to request that the host call the `work()`
+    /// method in a non-realtime context with the given arguments.
+    ///
+    /// This function is always safe to call from `run()`, but it is not guaranteed that the worker
+    /// is actually called from a different thread. In particular, when free-wheeling (e.g. for
+    /// offline rendering), the worker may be executed immediately. This allows single-threaded
+    /// processing with sample accuracy and avoids timing problems when `run()` is executing much
+    /// faster or slower than real-time.
+    ///
+    /// Plugins SHOULD be written in such a way that if the worker runs immediately, and responses
+    /// from the worker are delivered immediately, the effect of the work takes place immediately
+    /// with sample accuracy.
+    ///
+    /// **Notes about the passed data:** The buffer used to pass data is managed by the host. That
+    /// mean the size is unknown and may be limited. So if you need to pass huge amount of data,
+    /// it's preferable to use another way, for example a sync::mpsc channel.
+    pub fn schedule_work(&self, worker_data: P::WorkData) -> Result<(), WorkerError>
+    where
+        P::WorkData: 'static + Send,
+    {
+        unsafe {
+            let worker_data = mem::ManuallyDrop::new(worker_data);
+            let size = mem::size_of_val(&worker_data) as u32;
+            let ptr = &worker_data as *const _ as *const c_void;
+            let schedule_work = if let Some(schedule_work) = self.internal.schedule_work {
+                schedule_work
+            } else {
+                return Err(WorkerError::Unknown);
+            };
+            match (schedule_work)(self.internal.handle, size, ptr) {
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_SUCCESS => Ok(()),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_UNKNOWN => Err(WorkerError::Unknown),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_NO_SPACE => Err(WorkerError::NoSpace),
+                _ => Err(WorkerError::Unknown),
+            }
+        }
```

Comment:
You shouldn't use such a big `unsafe` block here, especially because there's only one unsafe operation in here. Also, you can shorten the expression to retrieve the `schedule_work` function ;)

```suggestion
        let worker_data = mem::ManuallyDrop::new(worker_data);
        let size = mem::size_of_val(&worker_data) as u32;
        let ptr = &worker_data as *const mem::ManuallyDrop<P::WorkData> as *const c_void;
        let schedule_work = self.internal.schedule_work.ok_or(WorkerError::Unknown)?;
        match unsafe { (schedule_work)(self.internal.handle, size, ptr) } {
            lv2_sys::LV2_Worker_Status_LV2_WORKER_SUCCESS => Ok(()),
            lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_UNKNOWN => Err(WorkerError::Unknown),
            lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_NO_SPACE => Err(WorkerError::NoSpace),
            _ => Err(WorkerError::Unknown),
        }
```

Reply:
solved in my local work